composeApp/src/main/kotlin/io/github/couchtracker/tmdb/TmdbDownload.kt

```diff
@@ -89,14 +94,24 @@ private suspend fun <T : Any> downloadAndSave(
     return element
 }
 
+/**
+ * @throws ApiException
+ */
+@Suppress("ThrowsCount", "SwallowedException")
 suspend fun <T> tmdbDownload(
     download: suspend (Tmdb3) -> T,
 ): T {
     try {
         return download(tmdb)
     } catch (e: CancellationException) {
         throw e
-    } catch (@Suppress("TooGenericExceptionCaught") e: Exception) {
-        throw TmdbException(e)
+    } catch (e: ClientRequestException) {
+        throw ApiException.ClientError(e.message, e)
+    } catch (e: ResponseException) {
+        throw ApiException.ServerError(e.message, e)
+    } catch (e: IOException) {
+        throw ApiException.IOError(e.message, e)
+    } catch (e: SerializationException) {
+        throw ApiException.IOError(e.message, e)
```

Comment:
I'm not sure that we should classify this as an IO error. Maybe we can add its own ApiException?